tests/models/test_model_settings.py

```diff
@@ -0,0 +1,199 @@
+
+    # Create wrapper - it should delegate to wrapped model's settings
+    wrapper = WrapperModel(base_model)
+    assert wrapper.settings == base_settings
+
+    # Test with wrapped model without settings
+    base_model_no_settings = TestModel()
+    wrapper_no_settings = WrapperModel(base_model_no_settings)
+    assert wrapper_no_settings.settings is None
+
+
+def test_instrumented_model_settings_delegation():
+    """Test that InstrumentedModel correctly delegates settings to wrapped model."""
+    # Create a base model with settings
+    base_settings = ModelSettings(max_tokens=100, temperature=0.5)
+    base_model = TestModel(settings=base_settings)
+
+    # InstrumentedModel should delegate settings to wrapped model
+    instrumented = InstrumentedModel(base_model)
+    assert instrumented.settings == base_settings
+
+    # Test with wrapped model without settings
+    base_model_no_settings = TestModel()
+    instrumented_no_settings = InstrumentedModel(base_model_no_settings)
+    assert instrumented_no_settings.settings is None
+
+
+def test_settings_merge_hierarchy():
+    """Test the complete settings merge hierarchy: model -> agent -> run."""
+    # Create a function that captures the merged settings
+    captured_settings = None
+
+    def capture_settings(messages: list[ModelMessage], agent_info: AgentInfo) -> ModelResponse:
+        nonlocal captured_settings
+        captured_settings = agent_info.model_settings
+        return ModelResponse(parts=[TextPart('captured')])
+
+    # Model settings (lowest priority)
+    model_settings = ModelSettings(max_tokens=100, temperature=0.5, top_p=0.8, seed=123)
+    model = FunctionModel(capture_settings, settings=model_settings)
+
+    # Agent settings (medium priority)
+    agent_settings = ModelSettings(
+        max_tokens=200,  # overrides model
+        temperature=0.6,  # overrides model
+        frequency_penalty=0.1,  # new setting
+    )
+    agent = Agent(model=model, model_settings=agent_settings)
+
+    # Run settings (highest priority)
+    run_settings = ModelSettings(
+        temperature=0.7,  # overrides agent and model
+        presence_penalty=0.2,  # new setting
+        seed=456,  # overrides model
+    )
+
+    # Run the agent
+    result = agent.run_sync('test', model_settings=run_settings)
+    assert result.output == 'captured'
+
+    # Verify the merged settings follow the correct precedence
+    assert captured_settings is not None
+    assert captured_settings['temperature'] == 0.7  # from run_settings
+    assert captured_settings['max_tokens'] == 200  # from agent_settings
+    assert captured_settings['top_p'] == 0.8  # from model_settings
+    assert captured_settings['seed'] == 456  # from run_settings
+    assert captured_settings['frequency_penalty'] == 0.1  # from agent_settings
+    assert captured_settings['presence_penalty'] == 0.2  # from run_settings
+
+
+def test_instrumented_model_settings_merge():
```

Comment:
I don't think we need this test anymore as there's no more conflict between instrumentation settings and model settings